lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -299,17 +299,30 @@ <h3 class="course-title" id="course-title-${enrollment.course_id}">
       %endif
 
       % if related_programs:
-      <div class="message message-related-programs is-shown">
-        <span class="related-programs-preface" tabindex="0">${_('Related Programs')}:</span>
-        <ul>
-          % for program in related_programs:
-          <li>
-            <span class="category-icon ${program['type'].lower()}-icon" aria-hidden="true"></span>
-            <span><a href="${program['detail_url']}">${u'{title} {type}'.format(title=program['title'], type=program['type'])}</a></span>
-          </li>
-          % endfor
-        </ul>
-      </div>
+        <div class="message message-related-programs is-shown">
+          <span class="related-programs-preface" tabindex="0">${_('Related Programs')}:</span>
+          <ul>
+            % for program in related_programs:
+            <li>
+              <span class="category-icon ${program['type'].lower()}-icon" aria-hidden="true"></span>
+              <span><a href="${program['detail_url']}">${u'{title} {type}'.format(title=program['title'], type=program['type'])}</a></span>
+            </li>
+            % endfor
+          </ul>
+          % if program_data.get('is_learner_eligible_for_one_click_purchase'):
+            <div class="complete-program-dashboard-div"><span class="complete-program-dashboard-span">Buy all remaining courses in this program and save 10%</span>
+              <a href="${urls.get('completeProgramURL')}" class="btn-brand btn cta-primary upgrade-button complete-program-dashboard-button">
+                ${_('Upgrade All Remaining Courses (')}
+                % if program_data.get('discount_data', {}).get('is_discounted'):
+                    <span class='list-price'>
+                      ${_('{"{0:.2f}".format(a)}'.format(program_data['discount_data']['total_incl_tax_excl_discounts'],2))}
+                    </span>
+                % endif
+                ${_(' ${price:.2f} {currency} ) '.format(price=program_data.get('full_program_price'), currency= program_data.get('discount_data', {}).get('currency')))}
```

Comment:
More formatting nits:
1. I'd put the arguments to format each on a new line.
2. Extra space after currency= .